src/enclave/App/App.cpp

```diff
@@ -277,104 +189,68 @@ JNIEXPORT jlong JNICALL Java_edu_berkeley_cs_rise_opaque_execution_SGXEnclave_St
 
   env->GetJavaVM(&jvm);
 
-  sgx_enclave_id_t eid;
-  sgx_launch_token_t token = {0};
-  int updated = 0;
+  oe_enclave_t* enclave = nullptr;
+  uint32_t flags = 0;
 
+#ifdef SIMULATE
+  flags |= OE_ENCLAVE_FLAG_SIMULATE;
+#endif
+  
   const char *library_path_str = env->GetStringUTFChars(library_path, nullptr);
-  sgx_check_and_time("StartEnclave",
-                     sgx_create_enclave(
-                       library_path_str, SGX_DEBUG_FLAG, &token, &updated, &eid, nullptr));
+  oe_check_and_time("StartEnclave",
+                     oe_create_Enclave_enclave(
+                       library_path_str, OE_ENCLAVE_TYPE_AUTO, flags, nullptr, 0, &enclave
+                      )
+                    );
   env->ReleaseStringUTFChars(library_path, library_path_str);
+  long int enclavePtr = (long int)enclave;
 
-  return eid;
-}
-
-JNIEXPORT jbyteArray JNICALL Java_edu_berkeley_cs_rise_opaque_execution_SGXEnclave_RemoteAttestation0(
-  JNIEnv *env, jobject obj, jlong eid) {
-  (void)obj;
-
-  sgx_status_t status;
-  sgx_check_and_time("Initialize Remote Attestation",
-                     ecall_enclave_init_ra(eid, &status, &context));
-  sgx_check("Initialize Remote Attestation", status);
-
-  uint32_t extended_epid_group_id = 0;
-  sgx_check_and_time("Remote Attestation Step 0: Get Extended EPID Group ID",
-                     sgx_get_extended_epid_group_id(&extended_epid_group_id));
-  jbyteArray ret = env->NewByteArray(sizeof(uint32_t));
-  env->SetByteArrayRegion(ret, 0, sizeof(uint32_t), (jbyte *) &extended_epid_group_id);
-  return ret;
+  return enclavePtr;
 }
 
-JNIEXPORT jbyteArray JNICALL Java_edu_berkeley_cs_rise_opaque_execution_SGXEnclave_RemoteAttestation1(
+JNIEXPORT jbyteArray JNICALL Java_edu_berkeley_cs_rise_opaque_execution_SGXEnclave_GenerateReport(
   JNIEnv *env, jobject obj, jlong eid) {
   (void)obj;
+  (void)eid;
 
-  sgx_ra_msg1_t msg1;
-  sgx_check_and_time("Remote Attestation Step 1",
-                     sgx_ra_get_msg1(context, eid, sgx_ra_get_ga, &msg1));
-  jbyteArray array_ret = env->NewByteArray(sizeof(sgx_ra_msg1_t));
-  env->SetByteArrayRegion(array_ret, 0, sizeof(sgx_ra_msg1_t), reinterpret_cast<jbyte *>(&msg1));
-  return array_ret;
-}
+  uint8_t* msg1 = NULL;
```

Comment:
Should we rename the variable from `msg1` to something else? The variable name is a bit confusing now that attestation doesn't involve 4 steps.

Reply:
Renamed the message names to something more readable.